Repeatable program stages

9sneha-n opened the pull request:
### :pushpin: References

-   **Issue:** Closes #?https://app.clickup.com/t/8693rygwd
- 8693rygwd

### :memo: Implementation
1. Remove unused custom repeat data element logic.
2. Fetch if a stage is repeatable from dhis api
3. For repeatable stages, show add/remove buttons in survey. 
4. Save repeated stages to DHIS
5. Parse repeatable stages and display for existing surveys. 

### :video_camera: Screenshots/Screen capture

https://github.com/EyeSeeTea/amr-surveys/assets/83749675/6896a4a2-5df2-4eef-b7a5-d955f87617e4


### :fire: Notes to the tester

anagperal commented on src/domain/usecases/GetSurveyUseCase.ts:
```diff
@@ -34,121 +34,127 @@ export class GetSurveyUseCase {
-                                                question.id === WARD_ID_DATAELEMENT_ID
-                                        ) !== undefined;
+            return this.getPPSSurveyForm(programId, parentPPSSurveyId, parentWardRegisterId);
+        } else if (parentPrevalenceSurveyId) {
```
Nice :)

anagperal commented on src/domain/usecases/GetSurveyUseCase.ts:
```diff
@@ -34,121 +34,127 @@ export class GetSurveyUseCase {
-                                                question.id === SURVEY_ID_PATIENT_DATAELEMENT_ID ||
-                                                question.id === WARD_ID_DATAELEMENT_ID
-                                        ) !== undefined;
+            return this.getPPSSurveyForm(programId, parentPPSSurveyId, parentWardRegisterId);
```
Nice!

anagperal commented on src/webapp/components/survey/SurveyForm.tsx:
```diff
@@ -79,9 +79,27 @@ export const SurveyForm: React.FC<SurveyFormProps> = props => {
     };
 
-    const updateQuestion = (question: Question) => {
+    const updateQuestion = (question: Question, stageId?: string) => {
```
This kind of functions: saveSurveyForm, updateQuestion, addProgramStage, removeProgramStage are logic and I think they should be placed in a hook

anagperal reviewed (changes requested):
Thanks @9sneha-n !! I have added some comments but they do not affect the functionality of the task, if code needs to be changed it can be done later if there is no time

9sneha-n commented:
> Thanks @9sneha-n !! I have added some comments but they do not affect the functionality of the task, if code needs to be changed it can be done later if there is no time

Thanks @anagperal  for the review, addressed the review comments.

anagperal reviewed (changes requested):
thanks @9sneha-n ! Just one last change, add `useCallback` to the functions of the hook to improve performance

9sneha-n commented:
> thanks @9sneha-n ! Just one last change, add `useCallback` to the functions of the hook to improve performance

Thank you, @anagperal, i missed that  . Have now added usecallbacks

anagperal reviewed (approved):
thanks @9sneha-n !

9sneha-n commented:
> thanks @9sneha-n !

Thanks @anagperal
